src/FlagProvider.tsx

```diff
@@ -106,20 +106,47 @@ const FlagProvider: FC<PropsWithChildren<IFlagProvider>> = ({
     };
   }, []);
 
+  const on = useCallback(
+    (event: string, callback: Function, ctx?: any) =>
+      client.current.on(event, callback, ctx),
+    []
+  ) as IFlagContextValue['on'];
```

Comment:
Maybe just wrapping a function works here?
```suggestion
  const on = useCallback<IFlagContextValue['on']>(client.current.on, []);
```

Reply:
@Tymek Hi sorry for the late update!, suggestions has been fixed please have a check